File: demo/ffi/ygui2/lua/hello.lua
-- ygui2 from Lua — the static hello. RUNNABLE inside a yetty pane:
--
--     LUA_PATH="bindings/lua/?.lua;;" luajit demo/ffi/ygui2/lua/hello.lua
--
-- One fullscreen frame over the drawable contract: a panel root, a column
-- of labels, a separator, a progress bar, and a statusbar. After the first
-- envelope the app sits idle — a clean frame ships ZERO bytes. Resize the
-- pane: the toolkit relays out and ships only the widgets whose geometry
-- actually changed. Ctrl-C quits (also `q` while no text input holds
-- focus).
local ygui2 = require("yetty.ygui2")

local app = ygui2.App()

local column = app.root:column{grow = 1, gap = 10, pad = 16}
column:label{text = "ygui2 — hello from Lua", fg = "#74C5A5", basis = 24}
column:separator{basis = 8}
column:label{text = "every widget is a wire group; this whole page was ONE envelope",
             basis = 20}
column:label{text = "clean frames ship zero bytes — watch the pane stay silent",
             fg = "#9FA7A8", basis = 20}
local row = column:row{basis = 24, gap = 10}
row:label{text = "progress", fg = "#9FA7A8", basis = 90}
row:progress{value = 0.42, basis = 220, cross = 12}
column:column{grow = 1.0} -- spacer pushes the statusbar to the bottom
column:statusbar{left = "hello.lua — static frame", right = "Ctrl-C: quit", basis = 24}

app:run()

File: demo/ffi/ygui2/go/hello.go
// ygui2 from Go — the static hello. RUNNABLE inside a yetty pane:
//
//	cd demo/ffi/ygui2/go
//	CGO_LDFLAGS="-L<build>/src/yetty/yffi -lyetty_ffi" \
//	LD_LIBRARY_PATH=<build>/src/yetty/yffi go run hello.go
//
// One fullscreen frame over the drawable contract: a panel root, a
// column of labels, a separator, a progress bar, and a statusbar. After
// the first envelope the app sits idle — a clean frame ships ZERO
// bytes. Resize the pane: the toolkit relays out and ships only the
// widgets whose geometry actually changed. Ctrl-C quits (also `q` while
// no text input holds focus).
package main

import (
	"fmt"
	"os"

	"github.com/zokrezyl/yetty/bindings/go/ygui2"
)

func main() {
	app := ygui2.NewApp()

	column := app.Root.Column(ygui2.Layout{Grow: 1, Gap: 10, Pad: 16})
	column.Label(ygui2.LabelOpts{Text: "ygui2 — hello from Go", Fg: "#74C5A5"},
		ygui2.Layout{Basis: 24})
	column.Separator(ygui2.Layout{Basis: 8})
	column.Label(ygui2.LabelOpts{
		Text: "every widget is a wire group; this whole page was ONE envelope"},
		ygui2.Layout{Basis: 20})
	column.Label(ygui2.LabelOpts{
		Text: "clean frames ship zero bytes — watch the pane stay silent", Fg: "#9FA7A8"},
		ygui2.Layout{Basis: 20})
	row := column.Row(ygui2.Layout{Basis: 24, Gap: 10})
	row.Label(ygui2.LabelOpts{Text: "progress", Fg: "#9FA7A8"}, ygui2.Layout{Basis: 90})
	row.Progress(ygui2.ProgressOpts{Value: 0.42}, ygui2.Layout{Basis: 220, Cross: 12})
	column.Column(ygui2.Layout{Grow: 1}) // spacer pushes the statusbar to the bottom
	column.Statusbar(ygui2.StatusbarOpts{Left: "hello.go — static frame",
		Right: "Ctrl-C: quit"}, ygui2.Layout{Basis: 24})

	if runError := app.Run(ygui2.RunOpts{}); runError != nil {
		fmt.Fprintln(os.Stderr, "hello:", runError)
		os.Exit(1)
	}
}
